packages/react/accordion/src/Accordion.tsx

```diff
@@ -37,46 +67,136 @@ type AccordionOwnProps = Polymorphic.Merge<
      */
     onValueChange?(value: string): void;
   }
 >;
 
-type AccordionPrimitive = Polymorphic.ForwardRefComponent<
-  Polymorphic.IntrinsicElement<typeof Primitive>,
-  AccordionOwnProps
+type AccordionSinglePrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof AccordionImpl>,
+  AccordionSingleOwnProps
 >;
 
-/**
- * `Accordion` is the root component.
- */
-const Accordion = React.forwardRef((props, forwardedRef) => {
+const AccordionSingle = React.forwardRef((props, forwardedRef) => {
   const {
     value: valueProp,
     defaultValue,
-    disabled,
     onValueChange = () => {},
-    ...accordionProps
+    ...accordionSingleProps
   } = props;
 
-  const buttonNodesRef = React.useRef<Set<React.ElementRef<typeof AccordionButton>>>(new Set());
-  const accordionRef = React.useRef<React.ElementRef<typeof Accordion>>(null);
-  const composedRefs = useComposedRefs(accordionRef, forwardedRef);
-
   const [value, setValue] = useControlledState({
     prop: valueProp,
     defaultProp: defaultValue,
-    onChange: (value) => value && onValueChange(value),
+    onChange: onValueChange,
+  });
+
+  return (
+    <AccordionValueProvider
+      value={value ? [value] : []}
+      onItemOpen={setValue}
+      onItemClose={() => setValue(undefined)}
+    >
+      <AccordionImpl {...accordionSingleProps} ref={forwardedRef} />
+    </AccordionValueProvider>
+  );
+}) as AccordionSinglePrimitive;
+
+type AccordionMultipleOwnProps = Polymorphic.Merge<
+  Polymorphic.OwnProps<typeof AccordionImpl>,
+  {
+    /**
+     * Allow mutltiple items to be open at the same time.
+     */
+    type: 'multiple';
+    /**
+     * The controlled stateful value of the accordion items whose panels are expanded.
+     */
+    value?: string[];
+    /**
+     * The value of the items whose panels are expanded when the accordion is initially rendered. Use
+     * `defaultValue` if you do not need to control the state of an accordion.
+     */
+    defaultValue?: string[];
+    /**
+     * The callback that fires when the state of the accordion changes.
+     */
+    onValueChange?(value: string[]): void;
+  }
+>;
+
+type AccordionMultiplePrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof AccordionImpl>,
+  AccordionMultipleOwnProps
+>;
+
+const AccordionMultiple = React.forwardRef((props, forwardedRef) => {
+  const {
+    value: valueProp,
+    defaultValue,
+    onValueChange = () => {},
+    ...accordionMultipleProps
+  } = props;
+
+  const [value = [], setValue] = useControlledState({
+    prop: valueProp,
+    defaultProp: defaultValue,
+    onChange: onValueChange,
   });
 
+  const handleItemOpen = React.useCallback(
+    (itemValue) => setValue((prevValue = []) => [...prevValue, itemValue]),
+    [setValue]
+  );
+
+  const handleItemClose = React.useCallback(
+    (itemValue) => setValue((prevValue = []) => prevValue.filter((value) => value !== itemValue)),
+    [setValue]
+  );
```

Comment:
This is a good example of why it's nice to have the callback implementation in the component that provides the context. I have the same context API but two different handler implementations 👍

Reply:
Love that